src/track/frame.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+class QDebug;
+
+namespace mixxx {
+class Frame {
+  public:
+    Frame()
+            : m_dFrame(0) {
+    }
+
+    explicit Frame(double dFrame)
+            : m_dFrame(dFrame) {
+    }
+
+    ~Frame() = default;
+
+    void setValue(double dFrame) {
+        m_dFrame = dFrame;
+    }
+
+    double getValue() const {
+        return m_dFrame;
+    }
+
+    Frame& operator+=(const Frame& arg) {
+        m_dFrame += arg.getValue();
+        return *this;
+    }
+
+    Frame& operator-=(const Frame& arg) {
+        m_dFrame -= arg.getValue();
+        return *this;
+    }
+
+    Frame& operator*=(const double arg) {
+        m_dFrame *= arg;
+        return *this;
+    }
+
+    Frame& operator/=(const double arg) {
+        m_dFrame /= arg;
+        return *this;
+    }
+
+    Frame operator++(int) {
+        Frame temp(*this);
+        m_dFrame++;
+        return temp;
+    }
+
+  private:
+    double m_dFrame;
+};
+
+// Frames can be added to and subracted from other frames (not by a double)
+inline const Frame operator+(const Frame& frame1, const Frame& frame2) {
```

Comment:
No, you cannot add a frame position to another frame position. But you could add a frame count to a frame position, resulting in a new position.

Using the *newtype* pattern for both positions **and** counts with well-defined arithmetic operations between them would be the correct solution. Let's first only encode the positions and use plain `double` for differences between positions. This is not type safe, but at least not wrong from a conceptual perspective.

Reply:
@daschuer Suggestions? Also related to my comment about the naming of the class `Frame`. 

Sure, pure double with an implicit interpretation would be even worse. But if we decide to define a type-safe class we need to be correct instead of introducing an incorrect abstraction.